### PR TITLE
Support .NET Core 3.1

### PR DESCRIPTION
**Pull request by vebbo2**

### CODE REVIEW

**Comment by vebbo2** [comment]
Hi @andrewlock , did you have a chance to take a look at this?

**Comment by andrewlock** [comment]
Hi @vebbo2, sorry for the delay! Thanks for this - just wondering, was there any reason this wasn't working previously? As far as I'm aware, this should be working on .NET Core 3.1 shouldn't it?

File: test/StronglyTypedId.Tests/StronglyTypedId.Tests.csproj [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFramework>netcoreapp3.1</TargetFramework>
```
Comment (andrewlock) [inline test/StronglyTypedId.Tests/StronglyTypedId.Tests.csproj]: This should work if we're building for .NET Core 3.1 shouldn't it?

Comment (vebbo2) [reply test/StronglyTypedId.Tests/StronglyTypedId.Tests.csproj]: Yes, it should be fine

**Comment by vebbo2** [comment]
@andrewlock Quoting you: https://github.com/andrewlock/StronglyTypedId/issues/5#issuecomment-524254501 :)
That's why CodeGeneration.Roslyn was upgraded and dotnet-codegen tool replaced with another package

File: test/StronglyTypedId.Tests/StronglyTypedId.Tests.csproj (hunk 1 above)
Comment (andrewlock) [reply test/StronglyTypedId.Tests/StronglyTypedId.Tests.csproj]: I mean, we want our tests to run on .netcore 3, but it looks like that broke the build? Ideally, I'd like the tests to target multiple platforms to confirm it all works correctly 🙂

**Comment by andrewlock** [comment]
Haha, of course 🤦‍♂️ thanks for jogging my memory! 🙂

File: test/StronglyTypedId.Tests/StronglyTypedId.Tests.csproj (hunk 1 above)
Comment (vebbo2) [reply test/StronglyTypedId.Tests/StronglyTypedId.Tests.csproj]: Oh, yes, AppVeyor seems to be configured to build with .NET Core 2.2. On my local machine it was obviously building correctly with .NET Core 3.1

Comment (andrewlock) [reply test/StronglyTypedId.Tests/StronglyTypedId.Tests.csproj]: Looks like appveyor is missing the `image` tag for some reason! 

If you add it in (as in here):
https://github.com/andrewlock/PwnedPasswords/blob/2e3f117bcd3939cd53e25ab4e1ca15cea918398e/appveyor.yml#L2

Then we should be able to update the tests to multi-target all supported runtimes to be safe 🙂

```xml
<TargetFrameworks>netcoreapp2.1,netcoreapp3.0,netcoreapp3.1</TargetFrameworks>
```

Thanks!

Comment (vebbo2) [reply test/StronglyTypedId.Tests/StronglyTypedId.Tests.csproj]: Done, build succeeded

Comment (andrewlock) [reply test/StronglyTypedId.Tests/StronglyTypedId.Tests.csproj]: Perfect, thanks!

**Comment by andrewlock** [comment]
Thanks @vebbo2, really appreciate you putting the effort in!😀

**Comment by andrewlock** [comment]
I was just looking at this again @vebbo2 - I think we should probably update `StronglyTyped` to use the suggested metapackage configuration, as described in https://github.com/AArnott/CodeGeneration.Roslyn#create-the-metapackage?

**Comment by vebbo2** [comment]
@andrewlock I didn't exactly follow with that example because main `StronglyTypedId` already contains references to Attributes and Generator projects through dependencies in nuspec file. But when I removed these dependencies and moved it to csproj (like in the example), build was failing because nuget package cannot be build if project contains neither any nuspec dependency nor any source file :/ That's why I left it as it is

**Comment by andrewlock** [comment]
@vebbo2 I've just made some tweaks in https://github.com/andrewlock/StronglyTypedId/pull/12 - Switched to using a 0-based enum for the Json type enum, and used the suggested metapackage config.

I'll push out a new release to NuGet shortly.

Thanks again for all your help! 😀

**Pull request merged**